lib/theme_check/language_server/server.rb

```diff
@@ -144,10 +206,16 @@ def read_new_content
       end
 
       def cleanup
+        # Stop listenting to RPC calls
+        Thread.kill(@json_rpc_thread) if @json_rpc_thread&.alive?
```

Comment:
I'm not sure how well this will work... If it ends up causing issues, you could call `@in.close` to unblock `@in.gets` similar to unblocking the queues w/ `queue.close`.

But just a gut feeling, might work OK here.

Reply:
Reworked and found a bug in the `read_new_content` loop. `@in.read(length + 2)` returns nil if @in is closed. It does not throw an error. This means that if it was impossible to read from STDIN, we entered an infinite loop.

Test code:
```rb
t = Thread.new do
  loop do
    # this code never throws, thus never terminates
    c = $stdin.read(5)
    puts "c", c.inspect
  end
end

sleep 1

$stdin.close

t.join
```

So I added a `raise DoneStreaming if @in.closed?` in there.

Good callout!